pyteal/ast/abi/util.py

```diff
@@ -205,3 +205,30 @@ def type_spec_from_annotation(annotation: Any) -> TypeSpec:
         return TupleTypeSpec(*(type_spec_from_annotation(arg) for arg in args))
 
     raise TypeError("Unknown annotation origin: {}".format(origin))
+
+
+T = TypeVar("T", bound=BaseType)
+
+
+def make(t: type[T]) -> T:
+    """Create a new instance of an ABI type. The type to create is determined by the input argument,
+    which must be a fully-specified annotation of type's class.
+
+    For example:
+        .. code-block:: python
+
+                # both of these are equivalent
+                myTuple = abi.make(abi.Tuple2[abi.Uint64, abi.DynamicArray[abi.Bool]])
+                myTuple = abi.Tuple(abi.TupleTypeSpec(abi.Uint64TypeSpec(), abi.DynamicArrayTypeSpec(abi.BoolTypeSpec())))
```

Comment:
totally optional...

Since this is expected to end up in the docs, I recommend adding this as a test case to `user_guide_test.py`. 

EG you could add something like "abi_make_snippet" to our [COPACETIC_SNIPPETS](https://github.com/algorand/pyteal/blob/master/tests/unit/user_guide_test.py#L63) with an assertions such as `assert myTuple_make == myTuple_explicit`

Reply:
I've opted not to do this for two reasons:
1. Existing user guide tests seem to be primarily concerned with comparing compiled TEAL code, and `make` does not affect this directly
2. `test_make` in `pyteal/ast/abi/util_test.py` already tests something very similar to the example. To fully close the loop, I updated the example to use the same expression as the test.